lib/renderer/content-scripts-injector.js

```diff
@@ -11,40 +11,54 @@ const matchesPattern = function (pattern) {
 }
 
 // Run the code with chrome API integrated.
-const runContentScript = function (extensionId, url, code) {
-  const context = {}
-  require('./chrome-api').injectTo(extensionId, false, context)
-  const wrapper = `(function (chrome) {\n  ${code}\n  })`
-  const compiledWrapper = runInThisContext(wrapper, {
-    filename: url,
-    lineOffset: 1,
-    displayErrors: true
-  })
-  return compiledWrapper.call(this, context.chrome)
+const runContentScripts = function (extensionId, js) {
+  const sandbox = {}
+  require('./chrome-api').injectTo(extensionId, false, sandbox)
+  Object.assign(sandbox, global)
+  const context = vm.createContext(sandbox)
+
+  let result = null
+  for (const {url, code} of js) {
```

Comment:
Concatenate all js files content on the fly and passing it to `runInThisContext` would work. (that's a similar method I implemented in [this workaround](https://github.com/getstation/repackage-chrome-extension-for-electron) and it works so far).

@zcbenz would you accept a PR with JS file concatenation on the fly?